plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/hover/HoverProvider.java

```diff
@@ -105,4 +127,89 @@ private String renderContent(Hover hover) {
     }
     return Joiner.on("\n\n").join(contents);
   }
+
+  private void createHoverLinkListener(EditorPartPresenter activeEditor) {
+    final Window window = Elements.getWindow();
+    window.addEventListener(
+        Event.MOUSEDOWN,
+        evt -> {
+          Element anchorEle = (Element) evt.getTarget();
+
+          // Register the onClick and open only if the element is in a tooltip.
+          if (anchorEle.getOffsetParent() != null
+              && anchorEle.getOffsetParent().getClassList() != null
+              && anchorEle.getOffsetParent().getClassList().contains("textViewTooltipOnHover")
+              && anchorEle.hasAttribute("href")) {
+            String hrefContent = anchorEle.getAttribute("href");
+            Location uriLocation = getLocationFromURI(hrefContent);
+            anchorEle.setOnclick(
+                anchorEleClick -> {
+                  anchorEleClick.preventDefault();
+                  anchorEleClick.stopPropagation();
+                  if (hrefContent.startsWith("file:/") || hrefContent.startsWith("jdt:/")) {
+                    openFileInEditorHelper.openLocation(uriLocation);
+                  } else {
+                    BrowserUtils.openInNewTab(hrefContent);
+                  }
+                  ((TextEditor) activeEditor).getEditorWidget().hideTooltip();
+                });
+          }
+        });
+  }
+
+  /**
+   * Create a workable URI for opening files in the workspace
+   *
+   * @param uri The URI to fix
+   * @return A URI that can be opened by the editor
+   */
+  private String createWorkableURI(String uri) {
+    String fixedURI = uri;
+    if (uri.indexOf("#") != -1) {
+      fixedURI = fixedURI.substring(0, uri.lastIndexOf("#"));
+    }
+
+    // Because of the markdown2html library in jdt.ls file:///projects gets stripped to
+    // file:/projects when converting the markdown to html
+    if (uri.startsWith("file:/projects")) {
```

Comment:
Here I guess we should consider both cases ("file:///" and "file:/") in order to make the code to process correctly both variants.

Reply:
No, if it's file:///, it's already correct and doesn't need to be changed.